### PR TITLE
Added verify method to RewriteRule

### PR DESCRIPTION
**Pull request by rdaly525**
Changed name of MapperSolution to RewriteRule.
Made RewriteRule take in minimally what it needs
Generalized some mapper.util functions
Added RewriteRule.verify() method

### CODE REVIEW

**Comment by rdaly525** [comment]
Fixing the CoreIR issue now

File: tests/test_aadt_mapping.py [hunk 1]
```diff
@@ -18,19 +20,86 @@ def test_automapper():
...
+
+    ir_fc = coreir_add_fc
+    arch_fc = lambda f: PE_fc(f)[0] #Only return peak class
+    output_binding = [(("out",), (0,))]
+
+    #Test correct rewrite rule
+    input_binding = [
+        (("in0",), ("a",)),
+        (("in1",), ("b",)),
+        (Inst_adt(Inst_bv.add)._value_, ("inst",)),
```
Comment (rdaly525) [inline tests/test_aadt_mapping.py:80]: Unfortunately this looks a bit gnarly. But this is currently how you get the Py BV value from the adt type.

Comment (cdonovick) [reply tests/test_aadt_mapping.py:79]: You could just assemble it by hand no?

Comment (rdaly525) [reply tests/test_aadt_mapping.py:79]: If you knew the exact bitvector (the case of Enum is easy) then yeah. Might be worth it to have a method on AADT that returns the underlying AbstractBitVector value

Comment (cdonovick) [reply tests/test_aadt_mapping.py:79]: I meant more just call `assembler.assemble(Inst_bv.add, BitVector)`  not sure if it's any more ergonomic.  Can't really add methods to AADTs without reserving field names so I would rather not.

**Pull request merged**